src/ui/flutter_app/lib/services/language_info.dart

```diff
@@ -116,7 +116,7 @@ String? specialCountryAndRegion;
 void setSpecialCountryAndRegion(BuildContext context) {
   final Locale currentLocale = Localizations.localeOf(context);
 
-  switch (currentLocale.countryCode) {
+  switch (currentLocale.toString()) {
```

Comment:
this should be changed to `currentLocale.languageCode`

My bad as I got confused with the countryCode/languageCode